package.json

```diff
@@ -54,7 +54,7 @@
 		"metro-resolver": "^0.70.3"
 	},
 	"scripts": {
-		"postinstall": "patch-package && npm ci --prefix gutenberg && npm run i18n:check-cache && ./bin/install-jetpack.sh",
+		"postinstall": "patch-package && npm run clean:gutenberg:distclean && npm ci --prefix gutenberg && npm run i18n:check-cache && ./bin/install-jetpack.sh",
```

Comment:
I'm wondering if it's necessary cleaning up Gutenberg on every dependencies install 🤔. I noticed in the `package-lock.json` that the `react` dependency has been updated to 18 ([reference](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5193/files#diff-053150b640a7ce75eff69d1a22cae7f0f94ad64ce9a855db544dda0929316519R8289-R8290)), shouldn't that be enough to install the correct versions of the dependencies under Gutenberg?

My main concern about this workaround is that I understand that it will increase severely the duration of the dependencies install command, do we know rough how much time do we expect that to be increased?

Reply:
Hey there, sorry for taking a bit to reply to this comment 😅 

>shouldn't that be enough to install the correct versions of the dependencies under Gutenberg?

It looks like we have other dependencies that still use React 17 and it's causing conflicts with the dependencies and the editor.

>My main concern about this workaround is that I understand that it will increase severely the duration of the dependencies install command, do we know roughly how much time we expect that to be increased?

What do you mean by: it will increase severely? That new step we are adding it's just removing any dependencies that might have been installed in Gutenberg when we run `npm install` in `Gutenberg mobile` before running `npm ci`. It's just a few seconds while it removes all of the `node_module` folders within `Gutenberg` that might have com from running `npm install` from the parent folder. From all of the tests I've done while working with the upgrade, it doesn't take more than a few seconds.

We could add a follow-up task to investigate further the dependencies we have to avoid this workaround.

Reply:
> It looks like we have other dependencies that still use React 17 and it's causing conflicts with the dependencies and the editor.

Ah, interesting. Do we know which dependencies are still using React 17?

On the other hand, do we need to run `npm run clean:gutenberg:distclean` everytime or is this a one-off?

> What do you mean by: it will increase severely? That new step we are adding it's just removing any dependencies that might have been installed in Gutenberg when we run npm install in Gutenberg mobile before running npm ci. It's just a few seconds while it removes all of the node_module folders within Gutenberg that might have com from running npm install from the parent folder. From all of the tests I've done while working with the upgrade, it doesn't take more than a few seconds.

Ah, ok, my concern came because I thought that cleaning Gutenberg would increase the duration of the `npm ci --prefix gutenberg` command. In case it's just adding a few seconds then it's ok to continue with this workaround. Thanks for elaborating on this 🙇 !

> We could add a follow-up task to investigate further the dependencies we have to avoid this workaround.

This would be great! Following on your comment regarding the fact that the duration won't be increased, I understand that is not a high priority, but having a ticket as a follow-up would be helpful.

Reply:
> Ah, interesting. Do we know which dependencies are still using React 17?

I saw a few but can't remember from when I tested, I'll add this in the ticket as well so we have a list.

> On the other hand, do we need to run npm run clean:gutenberg:distclean everytime or is this a one-off?

Just when an `npm install` happens, I know it's not ideal but we can work on this after this gets merged.

> This would be great! Following on your comment regarding the fact that the duration won't be increased, I understand that is not a high priority, but having a ticket as a follow-up would be helpful.

I mean it will be increased by a few seconds but it won't impact our development process or CI so much. But still, I'll create a ticket to investigate this a bit further.